src/Data/MultipartValue.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Saloon\Data;
+
+use InvalidArgumentException;
+use Saloon\Contracts\Arrayable;
+use Psr\Http\Message\StreamInterface;
+
+class MultipartValue implements Arrayable
+{
+    /**
+     * Constructor
+     *
+     * @param string $name
+     * @param \Psr\Http\Message\StreamInterface|resource|string|int $value
+     * @param string|null $filename
+     * @param array $headers
+     */
+    public function __construct(
+        public string  $name,
+        public mixed   $value,
+        public ?string $filename = null,
+        public array   $headers = []
+    ) {
+        if (! $this->value instanceof StreamInterface && ! is_resource($value) && ! is_string($value) && ! is_numeric($value)) {
```

Comment:
Just a small suggestion -
When I have plenty of checks, I prefer to split them and group them by context, and only invert at the beginning, with a group of 'or-checks' instead.
That way I don't have to put 'extra effort' to find all the inversions - I know that they have to be one or the other, as I group them.

```suggestion
        if (! ($this->value instanceof StreamInterface || is_resource($value) || is_string($value) || is_numeric($value))) {
```

Reply:
In this scenario, I'll keep `!` because it's readable enough